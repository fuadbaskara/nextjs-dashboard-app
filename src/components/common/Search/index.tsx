import React, { useState } from 'react';

export type SearchProps = {
  title: string;
  name: string;
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  value?: string;
  defaultValue?: string;
  placeholder?: string;
};

function Search({ title, name, onChange, value, defaultValue, placeholder }: SearchProps) {
  const [keyword, setKeyword] = useState('');

  return (
    <div>
      <label htmlFor={name} className="block text-sm font-medium text-gray-700">
        {title}
      </label>
      <div className="mt-1 flex rounded-md shadow-sm">
        <div className="relative flex flex-grow items-stretch focus-within:z-10">
          <input
            type="text"
            name={name}
            id={name}
            className="block w-full min-w-[200px] rounded-md border-gray-300 focus:border-indigo-500 focus:ring-indigo-500 sm:text-sm"
            autoComplete="off"
            value={value || keyword || defaultValue}
            placeholder={placeholder}
            onChange={(e) => {
              setKeyword(e.target.value);
              onChange(e);
            }}
          />
        </div>
      </div>
    </div>
  );
}

Search.defaultProps = {
  value: '',
  defaultValue: '',
  placeholder: '',
};

export default Search;
