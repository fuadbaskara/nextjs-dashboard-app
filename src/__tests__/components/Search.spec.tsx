/* eslint-disable testing-library/no-node-access */
import React from 'react';
import { render } from '../test-utils';
import Search from '@/components/common/Search';

const renderSearch = () =>
  render(
    <Search
      title="Search"
      name="search"
      placeholder="Search by keyword"
      value=""
      onChange={() => jest.fn()}
    />,
  );

describe('Search Component', () => {
  test('renders correctly with simple props:', () => {
    const { asFragment } = renderSearch();
    const search = document.querySelector('input[name="search"]');
    expect(search).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
