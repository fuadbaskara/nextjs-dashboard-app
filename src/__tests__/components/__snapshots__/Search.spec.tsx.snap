// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component renders correctly with simple props: 1`] = `
<DocumentFragment>
  <div>
    <label
      class="block text-sm font-medium text-gray-700"
      for="search"
    >
      Search
    </label>
    <div
      class="mt-1 flex rounded-md shadow-sm"
    >
      <div
        class="relative flex flex-grow items-stretch focus-within:z-10"
      >
        <input
          autocomplete="off"
          class="block w-full min-w-[200px] rounded-md border-gray-300 py-2 px-2 outline-none hover:cursor-pointer sm:text-sm"
          id="search"
          name="search"
          placeholder="Search by keyword"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
